http://sourceware.org/ml/gdb-cvs/2011-05/msg00112.html

### src/gdb/ChangeLog	2011/05/14 05:44:37	1.13009
### src/gdb/ChangeLog	2011/05/15 18:46:23	1.13010
## -1,3 +1,9 @@
+2011-05-15  Doug Evans  <dje@google.com>
+
+	* python/py-autoload.c (print_script): Print "Missing" instead of
+	"No" for missing scripts.
+	(info_auto_load_scripts): Tweak "Loaded" column to fit "Missing".
+
 2011-05-13  Doug Evans  <dje@google.com>
 
 	* ui-file.c (stdio_file_write_async_safe): Add comment.
--- src/gdb/python/py-auto-load.c	2011/05/13 22:11:47	1.9
+++ src/gdb/python/py-auto-load.c	2011/05/15 18:46:24	1.10
@@ -497,7 +497,7 @@
 
   chain = make_cleanup_ui_out_tuple_begin_end (uiout, NULL);
 
-  ui_out_field_string (uiout, "loaded", script->full_path ? "Yes" : "No");
+  ui_out_field_string (uiout, "loaded", script->full_path ? "Yes" : "Missing");
   ui_out_field_string (uiout, "script", script->name);
   ui_out_text (uiout, "\n");
 
@@ -569,7 +569,7 @@
   make_cleanup_ui_out_table_begin_end (uiout, 2, nr_scripts,
 				       "AutoLoadedScriptsTable");
 
-  ui_out_table_header (uiout, 6, ui_center, "loaded", "Loaded");
+  ui_out_table_header (uiout, 7, ui_left, "loaded", "Loaded");
   ui_out_table_header (uiout, 70, ui_left, "script", "Script");
   ui_out_table_body (uiout);
 
### src/gdb/doc/ChangeLog	2011/05/13 22:36:07	1.1190
### src/gdb/doc/ChangeLog	2011/05/15 18:46:25	1.1191
## -1,3 +1,7 @@
+2011-05-15  Doug Evans  <dje@google.com>
+
+	* gdb.texinfo (Auto-loading): Document printing of missing scripts.
+
 2011-05-13  Doug Evans  <dje@google.com>
 
 	* gdb.texinfo (Threads): Document $sdir,$pdir.
--- src/gdb/doc/gdb.texinfo	2011/05/13 22:36:07	1.838
+++ src/gdb/doc/gdb.texinfo	2011/05/15 18:46:25	1.839
@@ -23601,14 +23601,25 @@
 @kindex info auto-load-scripts
 @cindex print list of auto-loaded scripts
 @item info auto-load-scripts [@var{regexp}]
-Print the list of all scripts that gdb auto-loaded, or tried to auto-load.
+Print the list of all scripts that @value{GDBN} auto-loaded.
+
+Also printed is the list of scripts that were mentioned in
+the @code{.debug_gdb_scripts} section and were not found
+(@pxref{.debug_gdb_scripts section}).
+This is useful because their names are not printed when @value{GDBN}
+tries to load them and fails.  There may be many of them, and printing
+an error message for each one is problematic.
+
 If @var{regexp} is supplied only scripts with matching names are printed.
 
+Example:
+
 @smallexample
 (gdb) info auto-load-scripts
-Loaded Script                                                                 
-  Yes  py-section-script.py
-	full name: /tmp/py-section-script.py
+Loaded  Script
+Yes     py-section-script.py
+        full name: /tmp/py-section-script.py
+Missing my-foo-pretty-printers.py
 @end smallexample
 @end table
 
